### PR TITLE
Parallel vtk final

### PR DESCRIPTION
**Pull request by ptrscll**
This PR contains the latest version of the code for `_parallel_vtk.py` with condensed commits and clearer inline comments and variable names. 

The most substantial change left would be consolidating `run_particle_he` and `run_particle_ft` into one function and just putting some conditionals in the function in places where the code differs depending on the system being used. The main benefit to this would be that changes made to code  that is currently shared between functions would only need to be changed in one place instead of two. However, the combined function would be longer and would have to use more vague variable names like `run_vtk` does (e.g, we'd have to use "internal_vals" instead of "r" or "x", which could be more confusing). I don't **think** making this change would be too time-consuming to implement, since most of the code is the same to begin with, but I didn't want to make the change without discussing it first.

### CODE REVIEW

**Comment by dyvasey** [comment]
Awesome! I'll go through this later today. My inclination is to keep the functions separate and err on the side of readability.

One thought I had while working on notebooks - may want to add a `leave=False` flag to the `tqdm` call, so that when running lots of time steps the progress bar doesn't clutter the output.

**Review by dyvasey — CHANGES_REQUESTED** [review]
Peter, this looks awesome - great work! Especially on all the new tests. I don't think many changes are needed before merging. I made some suggested changes, mostly on the docstrings. And I wondered if we could come up with more specific exceptions for the calls to `suppress` to avoid letting bugs slip through. I was also just curious about the use of `np.inf` - don't see anything wrong with it but just wondering why that over `np.nan`.

File: gdtchron/_parallel_vtk.py [hunk 1]
```diff
@@ -0,0 +1,727 @@
...
+    age : float
+        Age of the particle. This equals np.nan if calc_age is False or there
+        is an issue obtaining the age of the particle
+    r : NumPy array of floats
+        Updated reduced lengths (unitless) for a hyppothetical grain located 
+        within this particle.
+
+    """
+    # Use float64 to match nans
+    dtype = np.float64
```
[the remark this answers is not in the document]
Comment (ptrscll) [reply gdtchron/_parallel_vtk.py:262]: I believe I was having issues using `np.float32` with `np.nan`s in the AFT code. However, when I tried switching it to `np.float32` just now, all the tests passed. My inclination would be to leave it as is in case we're missing some weird edge case in the tests, but I could also try switching the equivalent line in the thesis repo to `np.float32` and rerunning the post-processing script to see if it works.

[the remark this answers is not in the document]
Comment (dyvasey) [reply gdtchron/_parallel_vtk.py:263]: Yeah, let's just keep it as is for now - could be something to revisit if we want to work on optimization later - but FT already is a lot faster than He

File: gdtchron/_parallel_vtk.py [hunk 2]
```diff
@@ -0,0 +1,727 @@
...
+            # Find closest particle
+            distance, index = tree.query(particle_position)
+            
+            # Get id of closest particle
+            neighbor_id = tree_ids[index]
+            
+            # Get profile of closest particle
+            try:
+                r_initial = old_annealing_arrays[neighbor_id == old_ids][0]
+            except Exception:
```
[the remark this answers is not in the document]
Comment (dyvasey) [reply gdtchron/_parallel_vtk.py:319]: It's ok to leave it for now if it's going to be a headache to get the exact exception.

**Comment by dyvasey** [comment]
Yep, I think these can be future things to deal with. Next step is to get the notebooks running using the merged code, so that will be another test that things are working correctly.
